scripts/import/import_EPMC.pl

```diff
@@ -1265,8 +1278,23 @@ sub check_outdated_citations {
 
     my @split_attrib_id = split /,/, $attrib_id;
 
+    my @current_attribs;
     foreach my $attrib (@split_attrib_id) {
-      print $wrt "$variation_rsid\t$publication_pmid\t$attrib\n" unless ($citations_pheno_feature->{$attrib.'_'.$publication_pmid} || $citations_pheno_feature_attrib->{$attrib.'_'.$publication_pmid});
+      unless ($citations_pheno_feature->{$attrib.'_'.$publication_pmid} || $citations_pheno_feature_attrib->{$attrib.'_'.$publication_pmid}) {
+        push @current_attribs, $attrib;
+        print $wrt "$variation_rsid\t$publication_pmid\t$attrib\n";
+      }
+    }
+    
+    if (!@current_attribs) {
+      # remove citation if outdated in all sources
+      $rm_citations_sth->execute($attrib_id, $variation_id, $publication_id) ||
+        die "Error: cannot remove $variation_id, $publication_id, $attrib_id from variation_citation\n";
+    } elsif (@current_attribs ne @split_attrib_id) {
+      # discard outdated sources
+      my $new_attrib_id = join(",", @current_attribs);
+      $update_citations_sth->execute($new_attrib_id, $variation_id, $publication_id) ||
+        die "Error: cannot update $variation_id, $publication_id, $attrib_id from variation_citation with new value $new_attrib_id\n";
```

Comment:
Should we use `$variation_id` and `$publication_id` (more direct way to see where the problem is) or instead `$variaiton_rsid` and `$publication_pmid` as used for the output file?